chain/penumbra/penumbra_app_node.go

```diff
@@ -105,6 +106,61 @@ func (p *PenumbraAppNode) WalletPathContainer() string {
 	return filepath.Join(p.NodeHome(), "wallet")
 }
 
+func (p *PenumbraAppNode) ValidatorsInputFile() string {
+	return filepath.Join(p.Dir(), "validators.json")
+}
+
+func (p *PenumbraAppNode) ValidatorsInputFileContainer() string {
+	return filepath.Join(p.NodeHome(), "validators.json")
+}
+
+func (p *PenumbraAppNode) AllocationsInputFile() string {
+	return filepath.Join(p.Dir(), "allocations.csv")
+}
+
+func (p *PenumbraAppNode) AllocationsInputFileContainer() string {
+	return filepath.Join(p.NodeHome(), "allocations.csv")
+}
+
+func (p *PenumbraAppNode) GenesisFile() string {
+	return filepath.Join(p.Dir(), "node0", "tendermint", "config", "genesis.json")
+}
+
+func (p *PenumbraAppNode) ValidatorPrivateKeyFile(nodeNum int) string {
+	return filepath.Join(p.Dir(), fmt.Sprintf("node%d", nodeNum), "tendermint", "config", "priv_validator_key.json")
+}
+
+func (p *PenumbraAppNode) GenerateGenesisFile(
+	ctx context.Context,
+	chainID string,
+	validators []PenumbraValidatorDefinition,
+	allocations []PenumbraGenesisAppStateAllocation,
+) error {
+	validatorsJson, err := json.Marshal(validators)
+	if err != nil {
+		return fmt.Errorf("error marshalling validators to json: %v", err)
```

Comment:
`%w` is preferred and subtly different from `%v` for errors.

`%w` allows callers to use [`errors.Is`](https://pkg.go.dev/errors#Is) to introspect the wrapped error, whereas `%v` includes the string-formatted error in a new error, and the caller has no way to get at the wrapped error any more.

Reply:
Done